File: memory-domain/src/main/java/com/memory/domain/hashtag/HashTag.java
package com.memory.domain.hashtag;

import com.memory.domain.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.*;

@ToString
@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "hash_tag")
public class HashTag extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String name;

    @Column(nullable = false)
    private Long useCount = 0L;

    public HashTag(String name) {
        this.name = name;
        this.useCount = 0L;
    }

    public static HashTag create(String name) {
        return new HashTag(name);
    }

    public void incrementUseCount() {
        this.useCount++;
    }

    public void decrementUseCount() {
        if (this.useCount > 0) {
            this.useCount--;
        }
    }
}

File: memory-domain/src/main/kotlin/com/memory/domain/hashtag/HashTag.kt
package com.memory.domain.hashtag

import com.memory.domain.BaseTimeEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id

@Entity
class HashTag(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(unique = true, nullable = false)
    var name: String,

    @Column(nullable = false)
    var useCount: Long = 0L
) : BaseTimeEntity() {

    companion object {
        @JvmStatic
        fun create(name: String): HashTag = HashTag(name = name, useCount = 0L)
    }

    fun incrementUseCount() {
        this.useCount++
    }

    fun decrementUseCount() {
        if (this.useCount > 0) this.useCount--
    }
}